kedro/io/data_catalog.py

```diff
@@ -148,6 +148,7 @@ def __init__(  # noqa: too-many-arguments
         dataset_patterns: Patterns = None,
         load_versions: dict[str, str] = None,
         save_version: str = None,
+        raw_config: dict[str, Any] = None,
```

Comment:
I feel like there might be a way to do the resolving without saving the raw config here. We could use the config loader to get the raw catalog?

Reply:
+ the raw config in `DataCatalog` also has params which might need to be filtered out

Reply:
Added for documentation:

Spoke with @ankatiyar offline and whilst the `raw_config` will not have parameters that need filtering out, they will need to be filtered out of the pipeline entries to protect against a case where there is a catch-all pattern specified in the catalog.